services/workflows-service/src/transaction/transaction.controller.external.ts

```diff
@@ -333,42 +336,82 @@ export class TransactionControllerExternal {
     @Query() getTransactionsByAlertParameters: GetTransactionsByAlertDto,
     @CurrentProject() projectId: types.TProjectId,
   ) {
-    const alert = await this.prisma.alert.findUnique({
-      where: {
-        id: getTransactionsByAlertParameters.alertId,
+    const queryArgs = this.scopeService.scopeFindOne(
```

Comment:
You should use "scope service" in the repository, not in the service
the service should use the repository and it handles scoping etc..